go-controller/pkg/libovsdbops/db_index.go

```diff
@@ -0,0 +1,102 @@
+package libovsdbops
+
+import (
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+)
+
+const (
+	// db ownership ExternalIDs
+	// when possible, use DbIndex and function from this file instead of directly using the following ExternalIDs
+	OwnerController = "owner_controller"
+	OwnerObjectType = "owner_object_type"
+	OwnerNamespace  = "owner_namespace"
+	OwnerName       = "owner_name"
+	InternalID      = "internal_id"
+
+	// Owner Object Types
+	EgressFirewallDNSOwnerType = "EgressFirewallDNS"
+	EgressQoSOwnerType         = "EgressQoS"
+	NetworkPolicyOwnerType     = "NetworkPolicy"
+	NamespaceOwnerType         = "Namespace"
+	HybridNodeRouteOwnerType   = "HybridNodeRoute"
+)
+
+type DbIndex struct {
```

Comment:
This set of externalIDs is used to identify db objects, and it must be unique (which makes it semantically an index). I also hope to start using it as a primary and secondary index as you suggested in the previous comments. The docs are definitely should be improved (and hopefully will be, before this pr is merged)
I tried to define the limitations somewhat with e.g. `GetNoOwnerPredicate` function, that should only be used for the initial sync, otherwise the owner should always be set. I will try to update the comments a bit more about how to use dbIndex, if you have any other ideas for more strict verification, please lmk :)

Reply:
I understand the concept of the external ids themselves, but not how the `DbIndex` helps in this regard. Like, I see how it helps you handle the external ids. But I am not sure this is enough to constitute something more than that.
Let's say that I am proposing changing `DbIndex` name to `DbIndexHelper`.